compiler/compile/OMRCompilation.cpp

```diff
@@ -712,15 +712,17 @@ bool OMR::Compilation::isPotentialOSRPointWithSupport(TR::TreeTop *tt)
          if (node->getReferenceCount() > 1)
             {
             TR::TreeTop *cursor = tt->getPrevTreeTop();
-            TR::TreeTop *extendedBlockStart = tt->getNode()->getBlock()->startOfExtendedBlock()->getEntry();
-            while (cursor && cursor != extendedBlockStart)
+            while (cursor)
                {
                if ((cursor->getNode()->getOpCode().isCheck() || cursor->getNode()->getOpCodeValue() == TR::treetop)
                    && cursor->getNode()->getFirstChild() == node)
                   {
                   potentialOSRPoint = false;
                   break;
                   }
+               if (cursor->getNode()->getOpCodeValue() == TR::BBStart &&
+                   cursor == cursor->getNode()->getBlock()->startOfExtendedBlock()->getEntry())
```

Comment:
You want to test isExtensionOfPreviousBlock here - startOfExtendedBlock will walk back across blocks to return to you the treetop. You stop searching once BBStart says no to isExtensionOfPreviousBlock